src/behavioural-events/sdk-events.ts

```diff
@@ -64,60 +72,83 @@ export interface CheckoutSessionStartEvent extends IEvent {
   };
 }
 
-export interface CheckoutSessionFinishedEvent extends IEvent {
-  eventName: TrackedEventName.CheckoutSessionEnd;
+export interface CheckoutSessionFinishedEvent extends ISDKEvent {
+  eventName: SDKEventName.CheckoutSessionEnd;
   properties: {
     outcome: "finished";
     withRedemptionInfo: boolean;
   };
 }
 
-export interface CheckoutSessionClosedEvent extends IEvent {
-  eventName: TrackedEventName.CheckoutSessionEnd;
+export interface CheckoutSessionClosedEvent extends ISDKEvent {
+  eventName: SDKEventName.CheckoutSessionEnd;
   properties: {
     outcome: "closed";
   };
 }
 
-export interface CheckoutSessionErroredEvent extends IEvent {
-  eventName: TrackedEventName.CheckoutSessionEnd;
+export interface CheckoutSessionErroredEvent extends ISDKEvent {
+  eventName: SDKEventName.CheckoutSessionEnd;
   properties: {
     outcome: "errored";
     errorCode: number | null;
     errorMessage: string;
   };
 }
 
-export interface BillingEmailEntryImpressionEvent extends IEvent {
-  eventName: TrackedEventName.BillingEmailEntryImpression;
+export interface BillingEmailEntryImpressionEvent extends ISDKEvent {
+  eventName: SDKEventName.BillingEmailEntryImpression;
 }
 
-export interface BillingEmailEntryDismissEvent extends IEvent {
-  eventName: TrackedEventName.BillingEmailEntryDismiss;
+export interface BillingEmailEntryDismissEvent extends ISDKEvent {
+  eventName: SDKEventName.BillingEmailEntryDismiss;
 }
 
-export interface BillingEmailEntrySubmitEvent extends IEvent {
-  eventName: TrackedEventName.BillingEmailEntrySubmit;
+export interface BillingEmailEntrySubmitEvent extends ISDKEvent {
+  eventName: SDKEventName.BillingEmailEntrySubmit;
 }
 
-export interface BillingEmailEntrySuccessEvent extends IEvent {
-  eventName: TrackedEventName.BillingEmailEntrySuccess;
+export interface BillingEmailEntrySuccessEvent extends ISDKEvent {
+  eventName: SDKEventName.BillingEmailEntrySuccess;
 }
 
-export interface BillingEmailEntryErrorEvent extends IEvent {
-  eventName: TrackedEventName.BillingEmailEntryError;
+export interface BillingEmailEntryErrorEvent extends ISDKEvent {
+  eventName: SDKEventName.BillingEmailEntryError;
   properties: {
     errorCode: number | null;
     errorMessage: string;
   };
 }
 
-export interface PurchaseSuccessfulImpressionEvent extends IEvent {
-  eventName: TrackedEventName.PurchaseSuccessfulImpression;
+export interface PaymentEntryImpressionEvent extends ISDKEvent {
+  eventName: SDKEventName.PaymentEntryImpression;
 }
 
-export interface PurchaseSuccessfulDismissEvent extends IEvent {
-  eventName: TrackedEventName.PurchaseSuccessfulDismiss;
+export interface PaymentEntryDismissEvent extends ISDKEvent {
+  eventName: SDKEventName.PaymentEntryDismiss;
+}
+
+export interface PaymentEntryErrorEvent extends ISDKEvent {
+  eventName: SDKEventName.PaymentEntryError;
+  properties: {
+    stripeErrorCode: string | null;
+    stripeErrorMessage: string | null;
```

Comment:
Could we rename these fields to `gatewayError*` to make them more generic?
```suggestion
    gatewayErrorCode: string | null;
    gatewayErrorMessage: string | null;
```

Reply:
Yes, good call 👍🏻 !